src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -5667,4 +5668,193 @@ private <E extends Element> Void runUnderLint(E symbol, JCClassDecl p, BiConsume
 
     }
 
+    void checkRequiresIdentity(JCTree tree, Lint lint) {
+        switch (tree) {
+            case JCClassDecl classDecl : {
```

Comment:
Nit - unless fall-through is needed (it does not seem to be, on the first sight), I would suggest to use `->` instead of `:`, esp. given the cases use blocks anyway.